File: stash/vec_bench.c
#include <stdio.h>
#include "vec.h"

int main(void) {
  long score = 0;

  for (int i = 1; i < 50000; ++i) {
    CharVec vecc = vecc_new(i);
    IntVec veci = veci_new(i);

    for (int n = 0; n < 50; ++n) {
      vec_push(&vecc, (char)(n + 50));
      vec_push(&veci, n);
    }

    while (vecc.len && veci.len) {
      score += vec_wild_pop(&vecc);
      score += vec_wild_pop(&veci);
    }

    vec_free(&vecc);
    vec_free(&veci);
  }

  printf("%ld\n", score);
  
  return 0;
}

File: stash/vec_bench.rs
fn main() {
    let mut score: i64 = 0;
    let mut i = 1;

    while i < 50000 {
        let mut vecc = Vec::with_capacity(i);
        let mut veci = Vec::with_capacity(i);

        for n in 0..50 {
            vecc.push((n + 50) as u8);
            veci.push(n);
        }

        while 0 != vecc.len() && 0 != veci.len() {
            score += vecc.pop().unwrap() as i64;
            score += veci.pop().unwrap();
        }
        
        i += 1;
    }

    println!("{}", score);
}
